src/app/pages/courses-page/courses-page.component.ts

```diff
@@ -1,15 +1,110 @@
-import { Component, OnInit } from '@angular/core';
+import { AfterContentChecked, AfterContentInit, AfterViewChecked, AfterViewInit, Component, DoCheck, OnChanges, OnDestroy, OnInit } from '@angular/core';
+
+import { Course } from '../../models/course.model';
 
 @Component({
   selector: 'app-courses-page',
   templateUrl: './courses-page.component.html',
   styleUrls: [ './courses-page.component.scss' ],
 })
-export class CoursesPageComponent implements OnInit {
+export class CoursesPageComponent implements
+  OnChanges,
+  OnInit,
+  DoCheck,
+  AfterContentInit,
+  AfterContentChecked,
+  AfterViewInit,
+  AfterViewChecked,
+  OnDestroy {
+
+  public courses: Course[];
 
   constructor() { }
 
-  ngOnInit(): void {
+  ngOnChanges() {
+    // tslint:disable-next-line: no-console
+    console.log('OnChanges');
+  }
+
+  ngOnInit() {
+    // tslint:disable-next-line: no-console
+    console.log('OnInit');
+
+    this.populateCoursesArray();
+  }
+
+  ngDoCheck() {
+    // tslint:disable-next-line: no-console
+    console.log('DoCheck');
+  }
+
+  ngAfterContentInit() {
+    // tslint:disable-next-line: no-console
+    console.log('AfterContentInit');
+  }
+
+  ngAfterContentChecked() {
+    // tslint:disable-next-line: no-console
+    console.log('AfterContentChecked');
+  }
+
+  ngAfterViewInit() {
+    // tslint:disable-next-line: no-console
+    console.log('AfterViewInit');
+  }
+
+  ngAfterViewChecked() {
+    // tslint:disable-next-line: no-console
+    console.log('AfterViewChecked');
   }
 
+  ngOnDestroy() {
+    // tslint:disable-next-line: no-console
+    console.log('OnDestroy');
+  }
+
+  public onCourseDelete(courseId: number) {
+    // tslint:disable-next-line: no-console
+    console.log('Delete course with id: ', courseId);
+  }
+
+  private populateCoursesArray() {
+    this.courses = [ {
```

Comment:
It would be nice to have a `CoursesService` which would manage all data operations over courses

Reply:
Will do